site/cypress/e2e/Cypress-Gradeable/office_hours_queue.spec.js

```diff
@@ -0,0 +1,213 @@
+    if (queueCode.length > 0) {
+        cy.get('#old_queue_token').type(queueCode);
+    }
+    else {
+        cy.get('#old_queue_rand_token').click(); //random
+    }
+    cy.get('#change_code_btn').click(); // update it
+};
+const expectedAlerts =  (success=0, error=0, success_text='', error_text='') => {
+    if (success > 0) {
+        cy.get('.alert-success').contains(success_text);
+    }
+    if (error > 0) {
+        cy.get('.alert-error').contains(error_text);
+    }
+};
+const switchUser = (account) => {
+    cy.logout();
+    cy.login(account);
+    cy.visit(['sample']);
+    cy.get('#nav-sidebar-queue').should('exist').click();
+};
+const studentJoinQueue = (queueName, queueCode, name) => {
+    cy.get('#name_box').clear();
+    cy.get('#name_box').type(name);
+    cy.get('#queue_code').select(queueName).invoke('val'); // in which queue you want to join
+    cy.get('#queue_code').should('contain', queueName);
+    cy.get('#token-box').type(queueCode);
+    cy.get('#join_queue_btn').should('exist').click();
+};
+const studentRemoveSelfFromQueue = () => {
+    cy.get('#leave_queue').should('exist').click();
+};
+const studentFinishHelpSelf = () => {
+    cy.get('#self_finish_help').should('exist').click();
+};
+const helpFirstStudent = () => {
+    cy.get('.help_btn').first().should('exist').click();
+};
+const finishHelpFirstStudent = () => {
+    cy.get('.finish_helping_btn').first().should('exist').click();
+};
+const removeFirstStudent = () => {
+    cy.get('.remove_from_queue_btn').first().should('exist').click();
+};
+const restoreFirstStudent = () => {
+    cy.get('.queue_restore_btn').first().should('exist').click();
+};
+const toggleFirstQueueFilter = () => {
+    cy.get('.filter-buttons').first().should('exist').click();
+};
+const closeFirstQueue = () => {
+    cy.get('.toggle-queue-checkbox').first().should('exist').click();
+};
+const emptyFirstQueue = () => {
+    cy.get('.empty_queue_btn').first().should('exist').click();
+};
+const verifyElementMissing = (type, values) => {
+    values.forEach((value) => {
+        if (type === 'id') {
+            cy.get(`#${value}`).should('not.exist');
+        }
+        else if (type === 'class') {
+            cy.get(`.${value}`).should('not.exist');
+        }
+        else {
+            console.log(`Invalid ${type}`);
+        }
+    });
+};
+const openAnnouncementSettings = () => {
+    cy.get('#toggle_announcement_settings').click();
+    cy.get('#announcement-settings').should('exist');
+};
+const saveAnnouncementSettings = () => {
+    cy.get('#save_announcement').should('exist').click();
+};
+const editAnnouncement = (text='') => {
+    openAnnouncementSettings();
+    cy.get('#queue-announcement-message').clear();
+    if (text.length > 0) {
+        cy.get('#queue-announcement-message').type(text);
+    }
+    saveAnnouncementSettings();
+};
+describe('test office hours queue', () => {
+    it('test_office_hours_queue', () => {
```

Comment:
All of these can be one or two it statements, 8 is quite excessive.

Reply:
fixed